app/models/shipit/undeployed_commit.rb

```diff
@@ -35,12 +35,30 @@ def deploy_disallowed?
     end
 
     def deploy_discouraged?
-      stack.maximum_commits_per_deploy && index >= stack.maximum_commits_per_deploy
+      hit_maximum_commits_per_deploy?
+    end
+
+    def deploy_scheduled?
+      stack.continuous_deployment && !hit_maximum_commits_per_deploy? && !active?
     end
 
     def blocked?
       return @blocked if defined?(@blocked)
       @blocked = super
     end
+
+    def active?
+      return false unless stack.active_task?
+
+      active_task = stack.active_task
+
+      id > active_task.since_commit.id && id <= active_task.until_commit.id
+    end
+
+    private
+
+    def hit_maximum_commits_per_deploy?
```

Comment:
Not a huge fan of that naming. I'm not a good english speaker, but somehow `maximum_commits_per_deploy_hit?` or even `maximum_commits_per_deploy_reached?` would feel more natural to me.

However I defer to native speakers.

Reply:
I agree with you, I wasn't happy with the naming too, thanks!

Reply:
I like `maximum_commits_per_deploy_reached?` 🙂